loopdb/codec.go

```diff
@@ -1,8 +1,45 @@
 package loopdb
 
+import (
+	"fmt"
+
+	"github.com/lightninglabs/loop/looprpc"
+)
+
 // itob returns an 8-byte big endian representation of v.
 func itob(v uint64) []byte {
 	b := make([]byte, 8)
 	byteOrder.PutUint64(b, v)
 	return b
 }
+
+// MarshalProtocolVersion marshals a looprpc.ProtocolVersion value to a byte
+// slice.
+func MarshalProtocolVersion(version looprpc.ProtocolVersion) []byte {
```

Comment:
Server side we went for a separate enum for DB when storing protocol version. Reasoning was that a value not written in the DB isn't actually the same as a legacy rpc protcol version (they may be on preimage push version, for example, and we just weren't storing at that time). I was also concerned about a screw up on rpc level leaking through to the DB, although in practice it's unlikely to happen.

Reply:
Please note that using the proto generated enum is preexisting in the loop code.

In this change we don't fall back to the legacy version, but instead to `ProtocolVersion_USER_EXPIRY_LOOP_OUT` (which is the previous one) as the exact previous version does not matter as we only serialize for new swaps, and in that case the saved version will be `ProtocolVersion_HTLC_V2` (see the commit changing the current version).

The reasoning for not creating an extra enum wrapper over the proto generated enums is that it adds extra complexity where imo is not needed.

So to summarize:
- for old swaps: protocol version will be only used to decide between htlc v1/v2 scripts
- for newly created swaps: protocol version will be `ProtocolVersion_HTLC_V2` (see commit changing the current version for completeness)

Reply:
I strongly think it'd better to use proto generated enums but I also decided to instead go with the more common opinion and translate between RPC and DB enums to also keep the code more compatible with the server side. PTAL